src/pypi_simple/client.py

```diff
@@ -4,7 +4,7 @@
 from pathlib import Path
 import platform
 from types import TracebackType
-from typing import Any, AnyStr, Optional
+from typing import Any, AnyStr, Dict, Optional
```

Comment:
This file has `from __future__ import annotations`, so you can just use `dict` instead of `typing.Dict` (and I should probably replace the `Optional[T]`'s with `T | None` at some point too...).